unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/WorldChatWindowHUD/WorldChatWindowHUDController.cs

```diff
@@ -62,7 +57,33 @@ public void Dispose()
         Object.Destroy(view);
     }
 
-    int invalidSubmitLastFrame = 0;
+    void OnEnableWorldTab()
+    {
+        view.chatHudView.CleanAllEntries();
+
+        var result = chatController.GetEntries();
+
+        foreach (var v in result)
+        {
+            OnAddMessage(v);
+        }
+    }
+
+    void OnEnablePrivateTab()
+    {
+        view.chatHudView.CleanAllEntries();
+        var result = chatController.GetEntries().Where((x) => x.messageType == ChatMessage.Type.PRIVATE).ToList();
+
+        foreach (var v in result)
+        {
+            OnAddMessage(v);
+        }
```

Comment:
Can't we just hide/show the filtered messages by toggling their gameobjects? it would be more performant i think, also if we keep the private and non-private messages separately we would avoid traversing the whole entries list every time the tab is clicked.

Reply:
I don't think we need performance improvements on something so sparse as clicking on a tab, also this won't work well with this scenario:

- We have 30 private messages in 100 world chat entries (100 because its trimmed) but actually the private history is of 150.
- We change to private tab
- 30 messages would be shown, when with the 100 cap we could repopulate the list with 100 private messages.

Reply:
We could load the previous private messages at the beginning of the explorer and have the correct private messages in a separate collection, but OK we can tackle this if we notice problems